pkg/vault/engines.go

```diff
@@ -0,0 +1,39 @@
+package vault
+
+import (
+	"fmt"
+
+	"go.opencensus.io/trace"
+)
+
+// EnginesFromVault takes a path and returns a list of engines from vault.
+func (vc *vaultClient) EnginesFromVault(path string) ([]string, error) {
```

Comment:
Similar to what I ended up doing [here](https://github.com/teamsnap/vault-key/pull/28) you can keep this method as `enginesFromVault` so as not to export it as a Public method on the non exported struct vaultClient.

We'll want to create an exported function in vault.go that handles auth and creating a vault client and then calls this funcion.

Reply:
Makes sense. Updated this.